### PR TITLE
Add Active Support Instrumentation events

### PR DESCRIPTION
**Pull request by abicky**
`CronoTrigger::Events::MONITOR` helps us know there are unprocessed records for some reasons.
For example, the PR https://github.com/joker1007/crono_trigger/pull/35 caused some polling threads to sleep forever, and if we had monitored the information from this event, we could notice that something wrong was happening as soon as possible.
I also added `CronoTrigger::Events::PROCESS_RECORD`, which might be useful to record processing duration, count, etc.

### CODE REVIEW

File: lib/crono_trigger/worker.rb [hunk 1]
```diff
@@ -145,5 +157,37 @@ def handle_signal_from_rdb
...
+
+          next_execute_at_column = model.crono_trigger_column_name(:next_execute_at)
+          oldest_next_execute_at = model.executables.order(next_execute_at_column).limit(1).pluck(next_execute_at_column).first
+
+          now = Time.now
+          ActiveSupport::Notifications.instrument(CronoTrigger::Events::MONITOR, {
+            model_name: model_name,
+            executable_count: executable_count,
+            max_lock_duration_sec: oldest_execute_lock.nil? ? 0 : now.to_i - oldest_execute_lock,
+            max_latency_sec: oldest_next_execute_at.nil? ? 0 : now - oldest_next_execute_at,
```
Comment (joker1007) [inline lib/crono_trigger/worker.rb:185]: I think if all records have future time on `next_execute_at` column (that situation is a normal state), `max_latency_sec` returns a negative value.
Is that intentional?

Comment (abicky) [reply lib/crono_trigger/worker.rb:185]: If the next_execute_at of all records is a future time, I think the result is nil, and the latency will be 0. I'll add a test later for the case.

Comment (joker1007) [reply lib/crono_trigger/worker.rb:185]: Ah, `executable` scope rejects future `next_execute_at`. I missed that.
There is no problem. Thanks!

**Review by joker1007 — APPROVED** [review]
LGTM :+1:

File: lib/crono_trigger/worker.rb (hunk 1 above)
Comment (abicky) [reply lib/crono_trigger/worker.rb:185]: I added the test https://github.com/joker1007/crono_trigger/pull/39/commits/21a24be8b8e951f7f5697115086ec7aaf8ada5db.

**Pull request merged**